src/openrct2/ride/Track.cpp

```diff
@@ -910,9 +911,9 @@ bool track_remove_station_element(int32_t x, int32_t y, int32_t z, Direction dir
                     }
                     else
                     {
-                        ride->stations[stationIndex].Start.x = x;
-                        ride->stations[stationIndex].Start.y = y;
-                        ride->stations[stationIndex].Height = z;
+                        ride->stations[stationIndex].Start.x = currentLoc.x;
+                        ride->stations[stationIndex].Start.y = currentLoc.y;
+                        ride->stations[stationIndex].Height = loc.z;
```

Comment:
Possible bug in existing code. Need to investigate. Unsure why there is no multiply.